src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -369,6 +398,14 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
         }
         work_it = dispatch_queue.erase(work_it);
       } else {
+        sched_cls_info.running_tasks.insert(spec.TaskId());
```

Comment:
Hmm I find it confusing that we're setting the wait time here instead of at the same time as when the timer is checked above. Couldn't this cause a bug like this: CPU limit 1, task A gets scheduled and sets the wait timer. Task B gets queued much later on and gets scheduled immediately, when it should have waited for the timer.

Can we add a unit test to this effect?

Reply:
hmmm are you saying that the timer should only start when the raylet receives the next lease request? not when the capacity gets hit?

Reply:
btw, this also brings up the question, when does the timer start, if the lease request is submitted, but the task can't be dispatched because the arguments couldn't be pinned?

Reply:
I think the simplest would be to start the timer in the same spot where we check the timer.